File: Typescript/removeDuplicates.ts
function removeDuplicates(nums: number[]): number {
    if (nums.length === 0) {
        return 0;
    }

    let insertIdx = 1;
    for (let i = 1; i < nums.length; ++i) {
        if (nums[i - 1] !== nums[i]) {
            nums[insertIdx] = nums[i];
            insertIdx++;
        }
    }
    return insertIdx;
};

File: C++/removeDuplicates.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) 
    {
        if (nums.empty()) 
        {
            return 0;
        }

        int insertIdx = 1;
        for (int i = 1; i < nums.size(); ++i) 
        {
            if (nums[i - 1] != nums[i])
            {
                nums[insertIdx] = nums[i];
                insertIdx++;
            }
        }
        return insertIdx;
    }
};